packages/concerto-ui-react/src/concertoForm.js

```diff
@@ -114,12 +117,23 @@ class ConcertoForm extends Component {
     return { types, json };
   }
 
-  _loadAsyncData() {
-    return this.loadModelFiles(this.props.models, 'text');
+  async _loadAsyncData() {
+    this.setState({ loading: true });
+    const modelProps = await this.loadModelFiles(this.props.models, 'text');
+    this.setState({ loading: false });
+    return modelProps
   }
 
-  static getDerivedStateFromProps(props, state){
-    return { value: props.json, warning: null};
+  static getDerivedStateFromProps(props){
+    let shadowValue;
+    if(typeof props.json === 'string'){
+      try{
+        shadowValue = (JSON.parse(props.json));
+      } catch {}
```

Comment:
should `warning` be set in this `catch` block?

Reply:
Thanks, `warning` isn't actually used anymore, so I've removed it elsewhere too